File: src/main/kotlin/wowsft/model/gameparams/ship/component/airdefense/Aura.java
package wowsft.model.gameparams.ship.component.airdefense;

import wowsft.config.WoWSFT;
import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;

import java.util.List;

@Data
@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
public class Aura
{
    private float areaDamage;
    private float areaDamageFrequency;
    private float bubbleDamage;
    private float bubbleDuration;
    private float bubbleRadius;
    private boolean enableBarrage;
    private boolean enableCrewSelectedTargetCoeff;
    private float explosionCount;
    private List<String> guns;
    private float halfOuterBubbleZone;
    private float hitChance;
    private float innerBubbleCount;
    private List<Float> innerBubbleSpawnTimeRange;
    private float innerBubbleZone;
    @JsonAlias("isJoint")
    private boolean joint;
    private float maxBubbleActivationDelay;
    private float maxDistance;
    private float maxDistanceStartWorkGap;
    private float minBubbleActivationDelay;
    private float minDistance;
    private float outerBubbleCount;
    private List<Float> outerBubbleSpawnTimeRange;
    private float shotDelay;
    private float shotTravelTime;
    private float timeUniversalsOff;
    private float timeUniversalsOn;
    private String type;
    private float bubbleDamageModifier = 7.0f;
}

File: src/main/kotlin/wowsft/model/gameparams/ship/component/airdefense/Aura.kt
package wowsft.model.gameparams.ship.component.airdefense

import wowsft.config.WoWSFT
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonAlias
import wowsft.model.gameparams.ship.component.artillery.Shell

@WoWSFT
@JsonIgnoreProperties(ignoreUnknown = true)
class Aura
{
    var areaDamage = 0.0
    var areaDamageFrequency = 0.0
    var bubbleDamage = 0.0
    var bubbleDuration = 0.0
    var bubbleRadius = 0.0
    var enableBarrage = false
    var enableCrewSelectedTargetCoeff = false
    var explosionCount = 0.0
    var guns = ArrayList<String>()
    var halfOuterBubbleZone = 0.0
    var hitChance = 0.0
    var innerBubbleCount = 0.0
    var innerBubbleSpawnTimeRange = ArrayList<Double>()
    var innerBubbleZone = 0.0
    @JsonAlias("isJoint")
    var joint = false
    var maxBubbleActivationDelay = 0.0
    var maxDistance = 0.0
    var maxDistanceStartWorkGap = 0.0
    var minBubbleActivationDelay = 0.0
    var minDistance = 0.0
    var outerBubbleCount = 0.0
    var outerBubbleSpawnTimeRange = ArrayList<Double>()
    var shotDelay = 0.0
    var shotTravelTime = 0.0
    var timeUniversalsOff = 0.0
    var timeUniversalsOn = 0.0
    var type: String? = null
    var bubbleDamageModifier = 7.0
}
